[Core] AttributeError in compatibility patch causing test failures on WSL

### What happened + What you expected to happen

When running Ray unit tests on WSL, several tests are failing with an `AttributeError`. Specifically, these failures occur during the compatibility patch process, leading to a circular import issue.

Command executed:
```
python -m pytest -v -s python/ray/tests/test_basic.py
```

Observed Behavior:

The following test cases fail:

- test_default_worker_import_dependency
- test_omp_threads_set
- test_submit_api
- test_put_get
- test_wait_timing
- test_function_descriptor
- test_ray_options
- test_import_ray_does_not_import_grpc

Error message for reference:
```
       if (
            platform.system() == "Linux"
            and "Microsoft".lower() in platform.release().lower()
        ):
            import ray._private.compat  # noqa: E402

>           ray._private.compat.patch_psutil()
E           AttributeError: partially initialized module 'ray' has no attribute '_private' (most likely due to a circular import)

python/ray/__init__.py:60: AttributeError
```

Expected Behavior

The unit tests should pass without encountering an `AttributeError`.


Workaround

Changing the import statement from import `ray._private.compat` to `from ray._private import compat` in `ray/python/ray/__init__.py` may resolve this issue. This approach aligns with the import style of `ray._private.log` in the codebase and appears to circumvent the circular import problem.



### Versions / Dependencies

- Ray: 3.0 dev
- Linux: Ubuntu 22.04.3 LTS (Jammy)
- WSL: 5.15.137.3-microsoft-standard-WSL2

### Reproduction script

The issue arises when executing the following sequence in the unit tests.
This sequence should not cause an exception or a circular import problem.

```python
import ray
import sys
del sys.modules["ray"]
import ray
```


### Issue Severity

None

--- a/python/ray/__init__.py
+++ b/python/ray/__init__.py
@@ -55,9 +55,9 @@ def _configure_system():
         platform.system() == "Linux"
         and "Microsoft".lower() in platform.release().lower()
     ):
-        import ray._private.compat  # noqa: E402
+        from ray._private import compat  # noqa: E402
 
-        ray._private.compat.patch_psutil()
+        compat.patch_psutil()
 
     # Expose ray ABI symbols which may be dependent by other shared
     # libraries such as _streaming.so. See BUILD.bazel:_raylet

